bot/__init__.py

```diff
@@ -201,17 +201,27 @@ def clean_argument(arg: str) -> str:
             key = clean_argument(self.buffer[self.index:equals_pos])
 
             # Value: The second argument, specified after the `=`
-            value = (
-                clean_argument(
-                    self.buffer.split("=")[1]
+            rhs = self.buffer.split("=")[1].strip()
+
+            # If the value is None or '', mimick `bot.tags.delete(key)`
+            if rhs in ("None", "''", '""'):
+                log.trace(f"Command mimicks delitem. Key: {key!r}.")
+                result = self.buffer[self.previous:self.index] + ".delete"
+                args = f'"{key}"'
+
+            # Otherwise, assume assignment, for example `bot.tags['this'] = 'that'`
+            else:
+                value = (
+                    clean_argument(
+                        rhs
+                    )
+                    .replace("'", "\\'")  # escape any unescaped quotes
```

Comment:
this section is a bit crazy. I actually am not entirely sure how to clean this up.

possibly just like this?

```py
# escape any unescaped quotes
value = clean_argument(right_hand).replace("'", "\\'")
```

Reply:
I do admit I went a little overboard here. Fixed that.